backend/contract-service/src/main/java/org/egov/works/service/NotificationService.java

```diff
@@ -0,0 +1,336 @@
+        List<Estimate> estimates = estimateServiceUtil.fetchActiveEstimates(requestInfo, tenantId, lineItemsMap.keySet());
+        Map<String, String> projectDetails = projectServiceUtil.getProjectDetails(requestInfo, estimates.get(0));
+
+        Map<String,List<String>> orgDetails=organisationServiceUtil.getOrganisationInfo(request);
+        orgDetails.put("projectName", Collections.singletonList(projectDetails.get("projectName")));
+
+        return orgDetails;
+    }
+
+
+    private String getMessage(ContractRequest request, boolean isCBORole) {
+        //RETHINK-- NEED TO SEND TWO MESSAGES
+        Workflow workflow = request.getWorkflow();
+        String message = null;
+
+        if ("REJECT".equalsIgnoreCase(workflow.getAction()) && !isCBORole) {
+            message = getMessage(request, ContractServiceConstants.CONTRACTS_REJECT_LOCALIZATION_CODE);
+        } else if ("APPROVE".equalsIgnoreCase(workflow.getAction()) && !isCBORole) {
+            message = getMessage(request, ContractServiceConstants.CONTRACTS_APPROVE_CREATOR_LOCALIZATION_CODE);
+        } else if ("APPROVE".equalsIgnoreCase(workflow.getAction()) && isCBORole) {
+            message = getMessage(request, ContractServiceConstants.CONTRACTS_APPROVE_CBO_LOCALIZATION_CODE);
+
+        }
+
+        return message;
+    }
+
+    /**
+     * Gets the message from localization
+     *
+     * @param request
+     * @param msgCode
+     * @return
+     */
+    public String getMessage(ContractRequest request, String msgCode) {
+        String rootTenantId = request.getContract().getTenantId().split("\\.")[0];
+        Map<String, Map<String, String>> localizedMessageMap = getLocalisedMessages(request.getRequestInfo(), rootTenantId,
+                ContractServiceConstants.CONTRACTS_NOTIFICATION_ENG_LOCALE_CODE, ContractServiceConstants.CONTRACTS_MODULE_CODE);
+        return localizedMessageMap.get(ContractServiceConstants.CONTRACTS_NOTIFICATION_ENG_LOCALE_CODE + "|" + rootTenantId).get(msgCode);
+    }
+
+    /**
+     * Builds msg based on the format
+     *
+     * @param contract
+     * @param message
+     * @param userDetailsForSMS
+     * @return
+     */
+    public String buildMessageForRejectAction(Contract contract, Map<String, String> userDetailsForSMS, String message) {
+        message = message.replace("{CONTRACT_NUMBER}", contract.getContractNumber())
+                .replace("{PROJECT_NAME}", userDetailsForSMS.get("projectName"))
+                .replace("{LOCATION}", userDetailsForSMS.get("locationName"))
+                .replace("{USERNAME}", userDetailsForSMS.get("userName"))
+                .replace("{DESIGNATION}", userDetailsForSMS.get("designation"));
+        return message;
+    }
+
+    public String buildMessageForApproveAction_WO_CBO(Contract contract, Map<String, String> userDetailsForSMS, String message) {
+
+        long dueDate = System.currentTimeMillis() + TimeUnit.DAYS.toMillis(7);
```

Comment:
7 is hard coded can not be configuration driven

Reply:
changes done